### PR TITLE
docs: Reflect initial v2.x.x changes (closes #278)

### LINKED ISSUE #278: Update documentation to reflect Go project
Our [TeleIRC documentation](https://docs.teleirc.com) currently has mix-matched information between being a NodeJS and Go project. We need to change this to mirror our current status as a full Go project.

### PR DESCRIPTION
**Pull request by jwflory**
This commit makes a quick pass through all of our documentation to make
sure we clearly distinguish the project as a Go project, and not a Node
project. Some admonitions (i.e. `seealso::`, `note::`, `warning::`,
etc.) were added to mark some pages as works-in-progress, like our
deployment docs.

The contributor guidelines were also updated with basic guidance on how
to set up a developer environment for the Go project. Future docs PRs
will work on improving the things we have based on the v2.x.x changes.

Closes #278.

### CODE REVIEW

File: docs/config-file-glossary.rst [hunk 1]
```diff
@@ -17,11 +17,11 @@ IRC settings
 ``IRC_BOT_NAME=teleirc``
     IRC nickname for bot
 
-``IRC_CHANNEL=#channel``
+``IRC_CHANNEL="#channel"``
```
Comment (Tjzabel) [inline docs/config-file-glossary.rst]: ahhh, thanks for adding this.

File: docs/config-file-glossary.rst [hunk 2]
```diff
@@ -97,14 +97,7 @@ Imgur settings
...
-**********************
-Miscellaneous settings
-**********************
-
-``NTBA_FIX_319=1``
-    Required to fix a bug in a library used by TeleIRC.
-    For context, see `yagop/node-telegram-bot-api#319 <https://github.com/yagop/node-telegram-bot-api/issues/319#issuecomment-324963294>`_.
+``IMGUR_CLIENT_ID=7d6b00b87043f58``
+    Imgur API client ID value to access Imgur API.
+    Uses a default API key.
```
Comment (Tjzabel) [inline docs/config-file-glossary.rst]: Not a blocker, but maybe mentioning this is TeleIRC's general API key?

File: env.example [hunk 3]
```diff
@@ -16,7 +17,7 @@ IRC_PREFIX="<"
 IRC_SUFFIX=">"
 IRC_SERVER=chat.freenode.net
 IRC_SERVER_PASSWORD=""
-IRC_PORT=6667
+IRC_PORT=6697
```
Comment (Tjzabel) [inline env.example]: oohhh, thanks for the catch

File: docs/config-file-glossary.rst (hunk 2 above)
Comment (jwflory) [reply docs/config-file-glossary.rst]: @Tjzabel It's mentioned here: https://github.com/RITlug/teleirc/pull/279/files#diff-82fe77d984d3c73e3c8cd6c3216a3ce2R67

Do you think this is sufficient?

**Comment by jwflory** [comment]
@Tjzabel For what it's worth, our v1 docs are still published here too. :smile: 

https://docs.teleirc.com/en/v1.3.4/

**Pull request merged**